e2e/nomostest/reset.go

```diff
@@ -451,6 +451,7 @@ func initRepository(nt *NT, repoType gitproviders.RepoType, nn types.NamespacedN
 		}
 		nt.RemoteRepositories[nn] = repo
 	}
+	repo.WithLogger(nt.T, nt.Logger)
```

Comment:
how does this work? does it set? what happens to the new one returned?

Reply:
See the new method defined in `repository.go` - it uses the provided logger for the scope of the test and switches back to the original logger in cleanup

Reply:
Updated each of these to behave a bit more consistently

Reply:
This With- pattern doesn't follow the usual conventions. And the two different impls her have different semantics.

Usually I would expect a With- func to be a method on a builder struct with a Build() after that would return a new instance with the specified settings. 

The Repository one looks like a setter: Set-

The GitProvider one looks kind of like a an aspect oriented programming constructor: New-

But they both include cleanup unsetter, which is also weird.

Reply:
I'm going for a fixture type pattern here, where we set something for the context of the test and then restore it to the original state after the test is finished.

The difference in signatures was partly just for convenience, e.g. returning the pointer for GitProvider so that it can be inlined with the NT struct instantiation. I'm open minded on the name/signature if you have a suggestion

Reply:
I updated it to use a consistent `UseLogger` method for setting a logger for the scope of the test.

Reply:
Ok. I like Use- better than With-, just because it doesn't evoke a known pattern. But I think it's missing the detail that the setter has a scope: the duration of the current test. So what about just `SetTestLogger` or `UseLoggerForTest`? Then the method needs a comment about why it needs to be set and why it needs to be unset.

Reply:
SGTM

Reply:
The other idea that popped to mind was using AOP to wrap the struct with another struct that delegates to the first but uses a different logger. Then it could be unwrapped in cleanup, and the original object wouldn't have been modified. The problem with that is that I don't think Go's embedding allows overriding child methods the way Java inheritance does.

Reply:
done

Reply:
Hmm, since everything is using pointers we could probably just update the `Logger` itself to use the new `t` rather than assigning a new logger to everything

Reply:
That would be a lot cleaner if it works, let me give it a try

Reply:
yep looks like that works and is a lot simpler. PTAL

Reply:
much cleaner, thanks